packages/block-editor/src/store/private-actions.js

```diff
@@ -105,3 +113,188 @@ export function unsetBlockEditingMode( clientId = '' ) {
 		clientId,
 	};
 }
+
+/**
+ * Yields action objects used in signalling that the blocks corresponding to
+ * the set of specified client IDs are to be removed.
+ *
+ * Compared to `removeBlocks`, this private interface exposes an additional
+ * parameter; see `forceRemove`.
+ *
+ * @param {string|string[]} clientIds      Client IDs of blocks to remove.
+ * @param {boolean}         selectPrevious True if the previous block
+ *                                         or the immediate parent
+ *                                         (if no previous block exists)
+ *                                         should be selected
+ *                                         when a block is removed.
+ * @param {boolean}         forceRemove    Whether to force the operation,
+ *                                         bypassing any checks for certain
+ *                                         block types.
+ */
+export const privateRemoveBlocks =
+	( clientIds, selectPrevious = true, forceRemove = false ) =>
+	( { select, dispatch } ) => {
+		if ( ! clientIds || ! clientIds.length ) {
+			return;
+		}
+
+		clientIds = castArray( clientIds );
+		const rootClientId = select.getBlockRootClientId( clientIds[ 0 ] );
+		const canRemoveBlocks = select.canRemoveBlocks(
+			clientIds,
+			rootClientId
+		);
+
+		if ( ! canRemoveBlocks ) {
+			return;
+		}
+
+		// In certain editing contexts, we'd like to prevent accidental removal
+		// of important blocks. For example, in the site editor, the Query Loop
+		// block is deemed important. In such cases, we'll ask the user for
+		// confirmation that they intended to remove such block(s). However,
+		// the editor instance is responsible for presenting those confirmation
+		// prompts to the user. Any instance opting into removal prompts must
+		// register using `toggleRemovalPromptSupport()`.
+		//
+		// @see https://github.com/WordPress/gutenberg/pull/51145
+		if (
+			! forceRemove &&
+			// FIXME: Without this existence check, the unit tests for
+			// `__experimentalDeleteReusableBlock` in
+			// `packages/reusable-blocks/src/store/test/actions.js` fail due to
+			// the fact that the `registry` object passed to the thunk actions
+			// doesn't include this private action. This needs to be
+			// investigated to understand whether it's a real smell or if it's
+			// because not all store code has been updated to accommodate
+			// private selectors.
+			select.isRemovalPromptSupported &&
+			select.isRemovalPromptSupported()
```

Comment:
@jsnajdr: I quickly chatted with Adam about this. Is it possible that we are missing something in our Jest setup to accommodate private selectors?

Reply:
Thanks very much for the explanation! I agree that merging #39632 should be the way forward. In the meantime, if you like, I opened https://github.com/WordPress/gutenberg/pull/52088 for the short-term solution.